baseband/guppi/tests/test_guppi.py

```diff
@@ -0,0 +1,544 @@
+            assert (np.abs(fw.time - (start_time + self.spf / (2500 * u.Hz))) <
+                    1. * u.ns)
+
+        with guppi.open(filename, 'rs', squeeze=False) as fh:
+            data_onepol = fh.read()
+        assert np.all(data_onepol.squeeze() == self.payload[:self.spf, 0, :2])
+
+    # Test that writing an incomplete stream is possible, and that frame set is
+    # valid but invalid samples are appropriately marked.
+    def test_incomplete_stream(self, tmpdir):
+        filename = str(tmpdir.join('testguppi.raw'))
+        with catch_warnings(UserWarning) as w:
+            with guppi.open(filename, 'ws', header0=self.header_w,
+                            squeeze=False) as fw:
+                fw.write(self.payload[:10])
+        assert len(w) == 1
+        assert 'partial buffer' in str(w[0].message)
+        with guppi.open(filename, 'rs', squeeze=False) as fwr:
+            data = fwr.read()
+            assert np.all(data[:10] == self.payload[:10])
+            assert np.all(data[10:] == fwr.fill_value)
+
+    def test_multiple_files_stream(self, tmpdir):
+        start_time = self.header_w.time
+        with guppi.open(SAMPLE_FILE, 'rs') as fh:
+            data = fh.read()
+        filenames = (str(tmpdir.join('guppi_a.raw')),
+                     str(tmpdir.join('guppi_b.raw')))
+        with guppi.open(filenames, 'ws', header0=self.header_w,
+                        frames_per_file=2) as fw:
+            start_time = fw.start_time
+            fw.write(data[:1000])
+            time1000 = fw.time
+            fw.write(data[1000:])
+            stop_time = fw.time
+        assert start_time == self.header_w.time
+        assert np.abs(time1000 -
+                      (start_time + 1000 / (2500 * u.Hz))) < 1.*u.ns
+        assert np.abs(stop_time -
+                      (start_time + 30720 / (2500 * u.Hz))) < 1.*u.ns
+
+        with guppi.open(filenames[1], 'rs') as fr:
+            assert (np.abs(fr.time - (start_time + 15360 / (2500 * u.Hz))) <
+                    1. * u.ns)
+            data1 = fr.read()
+        assert np.all(data1 == data[15360:])
+
+        with guppi.open(filenames, 'rs') as fr:
+            assert fr.start_time == start_time
+            assert fr.time == start_time
+            assert np.abs(fr.stop_time -
+                          (start_time + 30720 / (2500 * u.Hz))) < 1. * u.ns
+            data2 = fr.read()
+            assert fr.time == fr.stop_time
+        assert np.all(data2 == data)
+
+        # Pass sequentialfile objects to reader.
+        with sf.open(filenames, 'w+b',
+                     file_size=(2 * self.header_w.frame_nbytes)) as fraw, \
+                guppi.open(fraw, 'ws', header0=self.header_w) as fw:
+            fw.write(data)
+
+        with sf.open(filenames, 'rb') as fraw, \
+                guppi.open(fraw, 'rs') as fr:
+            data3 = fr.read()
+        assert np.all(data3 == data)
+
+    # Test encoding and decoding frames and streams that use
```

Comment:
Just make this comment a docstring on the test method.

Reply:
I don't think we do this in general, but will make the change here and for `test_incomplete_stream`.

Reply:
I think it is good practice, especially if it explains what a whole method does rather than just a part of it (arguably, if we do different things, they should be different methods...  but no need to change anything...)